### DISCUSSION TITLE
CNAME is not supporting https

### DISCUSSION THREAD

**Discussion by djmenkveld** [spatie/laravel-google-cloud-storage#27] [Q&A]
In my setup I've a custom domain, https://assets.yosokugames.com/, attached to my bucket. A custom domain helps me to support https only. Now, following the readme I'm unable to configure a custom domain and you need to work with storage.googleapis.com, but in my case it is not helping, since my requirements is https.

Now, reading the Google docs, CNAME is only supporting HTTP requests:
https://cloud.google.com/storage/docs/request-endpoints#cname

Should I turn this question to a feature request for a custom domain?

**Comment by AlexVanderbist** [comment]
Hi @djmenkveld, if I understand correctly you have to use A records (and a Google Cloud load balancer with SSL) to use https:// enabled custom domains. In some of our client projects we're also using GCS like this. Without using the Google Cloud load balancer there simply isn't any service that will terminate the SSL connection for you.

For future reference, if you decide to got the A-record + GCLB way, here's our disk config that should work for custom domains:

```php
'gcs-disk' => [
        'key_file_path' => config_path('keys/google-service-account.json'),
        'storage_api_uri' => 'https://cdn.example.com/',
        'bucket' => 'example-bucket',
]
```

Good luck!
